src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1362,6 +1357,16 @@ void RaftPart::processAskForVoteRequest(
         }
     }
 
+    // If we have voted for somebody, we will reject other candidates under the proposedTerm.
+    if (votedAddr_ != std::make_pair(0, 0) && proposedTerm_ >= req.get_term()) {
```

Comment:
because we do not vote myself, so this check will not true if we are a candidate. and then the other candidate will convert we from candidate to follower